src/Core/Parser/Scene.cpp

```diff
@@ -111,6 +120,15 @@ int Raytracer::Scene::_parsePrimitiveSetting(const libconfig::Setting &primitive
                 std::shared_ptr<FlatColor> materialPtr = std::make_shared<FlatColor>(color["r"], color["g"], color["b"]);
                 newSphere->setMaterial(materialPtr);
             }
+
+            if (sphereArray[index].exists("translation")) {
+                libconfig::Setting& translation = sphereArray[index].lookup("translation");
+                Math::Vector3D trans(_parseValue(translation["x"]), _parseValue(translation["y"]), _parseValue(translation["z"]));
+                Math::Vector3D neworigin = newSphere->getOrigin();
+                neworigin.translate(trans);
+                newSphere->setOrigin(neworigin);
```

Comment:
Rename the variable to newOrigin.

```suggestion
                Math::Vector3D newOrigin = newSphere->getOrigin();
                newOrigin.translate(trans);
                newSphere->setOrigin(newOrigin);
```